doc/source/guide/spatial-partitioning.ipynb

```diff
@@ -0,0 +1,383 @@
+    "Currently, calculated distances along the curves are based on the mid points of geometries.\n",
+    "Whilst this is inexpensive, mid points do not represent complex geometries well.\n",
+    "Future work will examine various methods for better representating complex geometries.\n",
+    "\n",
+    "The Geohash, not to be confused with [geohashing](https://geohashing.site/geohashing/Main_Page) is a hierarchical spatial data structure, which subdivides space into buckets of grid shape.\n",
+    "Whilst the Geohash does allow binary searchers or spatial indexing, like rtree or quadtree, (a topic for later discussion & an enhancement for Dask-GeoPandas), Geohash is limited to encoding latitude and longitude coordinates.\n",
+    "The Geohash can be represented either as text string or integers, where the longer the string/integer is, the more precise the grid shape \n",
+    "will be.\n",
+    "\n",
+    "Below we compare the different sorting methods using the default parameters.\n",
+    "All 3 sorting methods reduce the degree of spatial overlap but this stil remains high and could be further improved by controlling specific parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0ad73df8-8264-4cd8-aa88-185d82ac5305",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hilbert = d_gdf.spatial_shuffle(by=\"hilbert\")\n",
+    "morton = d_gdf.spatial_shuffle(by=\"morton\")\n",
+    "geohash = d_gdf.spatial_shuffle(by=\"geohash\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1fcac318-06de-4144-9d34-b510bddda81e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(25,12))\n",
+    "ax1, ax2, ax3 = axes.flatten()\n",
+    "\n",
+    "for ax in axes:\n",
+    "    us_cont.plot(ax=ax)\n",
+    "\n",
+    "hilbert.spatial_partitions.plot(ax=ax1, cmap=\"tab20\", alpha=0.5)\n",
+    "morton.spatial_partitions.plot(ax=ax2, cmap=\"tab20\", alpha=0.5)\n",
+    "geohash.spatial_partitions.plot(ax=ax3, cmap=\"tab20\", alpha=0.5)\n",
+    "\n",
+    "[axi.set_axis_off() for axi in axes.ravel()]\n",
+    "\n",
+    "ax1.set_title(\"Hilbert\", size=16)\n",
+    "ax2.set_title(\"Morton\", size=16)\n",
+    "ax3.set_title(\"Geohash\", size=16)\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e34f773c-0eec-4a78-bc78-66dd6c4ebd22",
+   "metadata": {},
+   "source": [
+    "## npartitions\n",
+    "The first parameter is the `npartitions` or number of partitions.\n",
+    "The default npartitions is the same number of partitions as the original Dask GeoDataFrame.\n",
+    "Whilst increasing the number of partitions will reduce the degree of spatial overlap between partitions, this is also a more expensive operation.\n",
```

Comment:
> this is also a more expensive operation

But not by much right? This suggest significant difference while I don't think it will be so much more expensive.

Reply:
With the us example provided there isn't much overhead going from 4 partitions to 10 partitions.
However, there is more overhead going from 10 to 100 partitions with natural earth low reg (~110ms vs 1s on my machine);

```python
import geopandas
import dask_geopandas

df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
ddf = dask_geopandas.from_geopandas(df, 4)

%timeit ddf.spatial_shuffle(npartitions=10)
%timeit ddf.spatial_shuffle(npartitions=100)
```
My understanding is that rearranging your DataFrame into smaller chunks incurrs more overhead than arranging it into fewer ones.
This should better link with Breaking your DataFrame into "too many" partitions results in [the scheduler incurring more overhead deciding where to compute each task
So it's a cost of creating a Dask GeoDataFrame vs cost of computation with each following task.

Reply:
What you see is mostly the scheduler overhead. With GeoDataFrames of a larger size, that is minimised.

```py
import geopandas
import dask_geopandas
import pandas

df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
ddf = dask_geopandas.from_geopandas(pandas.concat([df] * 1000), 4)

%timeit ddf.spatial_shuffle(npartitions=10)
%timeit ddf.spatial_shuffle(npartitions=100)

1.25 s ± 72 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
1.41 s ± 31.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Gotcha, I should have tested this with a larger dataset.
FYI, my results were;
```
1.54 s ± 36.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
2.63 s ± 237 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
which still isn't much - the most important thing to consider then is the scheduler incurring overhead, when deciding where to compute each subsequent task.
I'll update the notebook.